src/main/kotlin/org/jitsi/nlj/stats/EndpointConnectionStats.kt

```diff
@@ -85,22 +90,23 @@ class EndpointConnectionStats(
         when (packet) {
             is RtcpSrPacket -> {
                 logger.cdebug { "Tracking sent SR packet with compacted timestamp ${packet.senderInfo.compactedNtpTimestamp}" }
-                srSentTimes[Pair(packet.senderSsrc, packet.senderInfo.compactedNtpTimestamp)] =
-                        System.currentTimeMillis()
+                srSentTimes[Pair(packet.senderSsrc, packet.senderInfo.compactedNtpTimestamp)] = clock.instant()
             }
         }
     }
 
-    private fun processReportBlock(receivedTime: Long, reportBlock: RtcpReportBlock) {
+    private fun processReportBlock(receivedTime: Instant, reportBlock: RtcpReportBlock) {
         if (reportBlock.lastSrTimestamp > 0 && reportBlock.delaySinceLastSr > 0) {
             // We need to know when we sent the last SR
-            val srSentTime = srSentTimes.getOrDefault(SsrcAndTimestamp(reportBlock.ssrc, reportBlock.lastSrTimestamp), -1)
-            if (srSentTime > 0) {
+            srSentTimes[SsrcAndTimestamp(reportBlock.ssrc, reportBlock.lastSrTimestamp)]?.let { srSentTime ->
                 // The delaySinceLastSr value is given in 1/65536ths of a second, so divide it by 65.536 to get it
                 // in milliseconds
                 val remoteProcessingDelayMs = reportBlock.delaySinceLastSr / 65.536
-                rtt = receivedTime - srSentTime - remoteProcessingDelayMs
+                rtt = receivedTime.toEpochMilli() - srSentTime.toEpochMilli() - remoteProcessingDelayMs
                 endpointConnectionStatsListeners.forEach { it.onRttUpdate(rtt) }
```

Comment:
Do you want to log if you get surprisingly high rtt values?

Reply:
yeah, will add this